### ISSUE TITLE
Training consumption

### ISSUE THREAD

**Issue by sms-s** [GAD-cell/VLM_GRPO#1]
Your work looks amazing! I'm curious about the training data size, GPU configuration, and GPU usage for your GRPO. Also, how significant is the difference in results between GRPO with and without unsolth? I'm looking forward to your response.

**Comment by GAD-cell** [comment]
Hey, thanks for your comment!

GPU usage largely depends on your input,especially with VVLM since it mainly varies based on the image size and the number of images you provide.

With GRPO, you also need to generate n outputs per query to compute a loss, which can quickly multiply GPU usage.

For example: with a single 512×512 image and num_generation=4, I use around 30 GB of VRAM on an A100 with Qwen2.5 VL 7B.

As for training data size, it depends on your setup, but a typical GRPO setup might involve few thousands samples, though some studies suggest that you can do LLM alignements with few shots training. 

Using unsloth mainly helps reduce memory usage and training time thanks to gradient checkpointing and LoRA. With the same LoRA config, you shouldn’t see any significant difference in results between using unsloth and not.

The main benefit of the VLM GRPO implementation for unsloth is that it makes quick experiments easier and friendly, great for doing some fast and “quick and dirty” testing 😄

Hope it'll help you !

**Comment by sms-s** [comment]
Thank you for your reply. Have you made a comparison with full-parameter GRPO? Is there any difference in the training results?

> 嘿，谢谢你的评论！
> 
> GPU 的使用很大程度上取决于您的输入，尤其是 VLM，因为它主要根据图像大小和您提供的图像数量而变化。
> 
> 使用 GRPO，您还需要为每个查询生成 n 个输出来计算损失，这会迅速增加 GPU 的使用率。
> 
> 例如：对于单个 512×512 图像和 num_generation=4，我在带有 Qwen2.5 VL 7B 的 A100 上使用大约 30 GB 的 VRAM。
> 
> 至于训练数据的大小，这取决于您的设置，但典型的 GRPO 设置可能涉及几千个样本，尽管一些研究表明您可以通过几次训练进行 LLM 比对。
> 
> 使用 unsloth 主要有助于减少内存使用量和训练时间，这要归功于梯度检查点、LoRA 量化......使用相同的 LoRA 配置，您不应该看到使用 unsloth 和不使用 unsloth 之间的结果有任何显著差异。
> 
> VLM GRPO 实现对 unsloth 的主要好处是，它使快速实验变得更容易、更友好，非常适合进行一些快速且“快速而粗糙”的测试😄
> 
> 希望它能帮到你！

Thank you for your reply. Have you made a comparison with full-parameter GRPO? Is there any difference in the training results?

**Comment by GAD-cell** [comment]
I did not, I don't have enough Vram for this. It should be different yes, you might also more easily overfit the model, so Lora is still a better option I think rather than training all the parameters as you will never get as rich datas as the ones it's been pre-trained with.

**Comment by sms-s** [comment]
Okay, thank you.
